File: backend/src/main/java/com/my/portfolio/domain/ports/application/ProjectService.java
package com.my.portfolio.domain.ports.application;

import com.my.portfolio.api.v1.model.ProjectDto;
import com.my.portfolio.api.v1.model.ProjectSummaryDto;
import java.util.List;

public interface ProjectService {

  ProjectDto getProjectById(long id);

  List<ProjectSummaryDto> getProjectsSummaries();
}

File: my-portfolio-frontend/src/app/core/services/project.service.ts
import { Injectable } from '@angular/core';
import { Observable } from '../../../../node_modules/rxjs/dist/types/index';
import { ProjectDto, ProjectsOpenApiService, ProjectSummaryDto } from '../../../generated-api-client-v1/index';


@Injectable({
  providedIn: 'root'
})
export class ProjectService {

  constructor(private projectsOpenApiService: ProjectsOpenApiService ) { }

  public getProjectById(id : number) : Observable<ProjectDto> {
    return this.projectsOpenApiService.getProjectById(id);
  }

  public getProjectsSummaries() : Observable<Array<ProjectSummaryDto>> {
    return this.projectsOpenApiService.getProjectsSummaries();
  }

}
